main/src/ca/uwaterloo/flix/language/phase/Typer.scala

```diff
@@ -1604,14 +1604,14 @@ object Typer {
         for {
           (constrs, tpe, _, eff) <- visitExp(exp)
           _ <- expectTypeM(expected = Type.Int32, actual = tpe, exp.loc)
-          resultTyp <- liftM(Type.mkTuple(List(Type.mkSender(elmType, loc), Type.mkReceiver(elmType, loc)), loc))
+          resultTyp <- liftM(Type.mkTuple(List(Type.mkSender(elmType, Type.False, loc), Type.mkReceiver(elmType, Type.False, loc)), loc))
```

Comment:
Here you should not use Type.False but create a fresh type variable and unify with it.

Reply:
For this and the following similar comments, I see why this would be the right thing to do, but am I not going to then have to immediately discard that code as soon as I add a region parameter to the channel functions (because I'll be unifying against that parameter and won't have to create one here?)

Or am I misunderstanding?

Reply:
It would be the opposite. (The change I propose would be forwards compatible).

The point is that only GetChannel takes a region parameter. For the other channel operations the region is inferred by the type system (hence the need for type variables that get unified.)

Reply:
Oh! Then I am definitely misunderstanding. 

Do you mean GetChannel? Or NewChannel? Above you were talking about adding a `r: Region[r]` parameter to `Channel.buffered`?

Forgive me, I'm confused.

Reply:
Maybe I am confused. My point was for everything but NEW_CHANNEL we should infer the region (i.e. not use Static/False, but rather a type variable).

Reply:
Here I am only talking about what occurs in Typer and only the AST nodes.

Reply:
Ah, NewChannel. That makes much more sense. OK, I think I get it, thanks.